File: sprint6/By languages/C#/lesson6code1.cs
using System.Collections.Generic;

Stack<int> order = new Stack<int>(); // В этом стеке будет записан порядок обхода.
int[] color = new int[n];

void topSort(int v) {
    color[v] = gray;
    foreach (int w in adjList[v]) {
        if (color[w] == white) {
            topSort(w);
        }
    }
    color[v] = black;
    order.Push(v); // Кладём обработанную вершину в стек.
}

void mainTopSort() {
    for (int i = 0; i < n; i++) {
        if (color[i] == white) {
            topSort(i);
        }
    }
}

File: sprint6/By languages/kotlin/lesson6code1.kt
val order = Stack<Int>() // В этом стеке будет записан порядок обхода.
val color = Array(size = n) { white }

fun topSort(v: Int) {
    color[v] = gray
    for (w in adjList[v]) {
        if (color[w] == white) {
            topSort(w)
        }
    }
    color[v] = black
    order.push(v) // Кладём обработанную вершину в стек.
}

fun mainTopSort() {
    for (i in 0 until n) {
        if (color[i] == white) {
            topSort(i)
        }
    }
}
